spec/lib/disability_compensation/providers/document_upload/lighthouse_supplemental_document_upload_provider_spec.rb

```diff
@@ -207,55 +207,50 @@
-        {
-          'errors' => [
-            {
-              'detail' => 'Something broke',
-              'status' => 400,
-              'title' => 'Bad Request',
-              'instance' => Faker::Internet.uuid
-            }
-          ]
-        }
-      end
-
+    # See lib/lighthouse/service_exception.rb
+    context 'when a Lighthouse::ServiceException error is raised' do
       before do
         # Skip upload attempt logging
         allow(provider).to receive(:log_upload_attempt)
+      end
 
-        allow(BenefitsDocuments::Form526::UploadSupplementalDocumentService).to receive(:call)
-          .with(file_body, lighthouse_document)
-          .and_return(faraday_response)
+      RSpec.shared_examples 'log Lighthouse response exception' do |exception_class|
+        it 'increments a StatsD failure metric, logs the error metadata and re-raises the error' do
+          error_info = { title: 'error title', detail: 'error message' }
+          exception = exception_class.new(error_info)
 
-        allow(faraday_response).to receive(:body).and_return(error_response_body)
-      end
+          allow(BenefitsDocuments::Form526::UploadSupplementalDocumentService).to receive(:call)
+            .and_raise(exception)
 
-      it 'logs to the Rails logger' do
-        expect(Rails.logger).to receive(:error).with(
-          'LighthouseSupplementalDocumentUploadProvider upload failed',
-          {
-            class: 'LighthouseSupplementalDocumentUploadProvider',
-            submitted_claim_id: submission.submitted_claim_id,
-            submission_id: submission.id,
-            user_uuid: submission.user_uuid,
-            va_document_type_code: va_document_type,
-            primary_form: 'Form526',
-            lighthouse_error_response: error_response_body
-          }
-        )
+          expect(StatsD).to receive(:increment).with(
+            'my_stats_metric_prefix.lighthouse_supplemental_document_upload_provider.upload_failure'
+          )
+          expect(Rails.logger).to receive(:error).with(
+            'LighthouseSupplementalDocumentUploadProvider upload failed',
+            {
+              class: 'LighthouseSupplementalDocumentUploadProvider',
+              submitted_claim_id: submission.submitted_claim_id,
+              submission_id: submission.id,
+              user_uuid: submission.user_uuid,
+              va_document_type_code: va_document_type,
+              primary_form: 'Form526',
+              error_info: error_info.to_s
+            }
+          )
 
-        provider.submit_upload_document(lighthouse_document, file_body)
+          expect { provider.submit_upload_document(lighthouse_document, file_body) }.to raise_error(exception)
+        end
       end
 
-      it 'increments a StatsD metric' do
-        expect(StatsD).to receive(:increment).with(
-          'my_stats_metric_prefix.lighthouse_supplemental_document_upload_provider.upload_failure'
-        )
+      describe 'service exceptions' do
```

Comment:
I haven't yet been able to get all of these tests passing with the Rails logger logging, some work and some do not which is confusing since they are all just custom exception classes. The metrics increment and re-raising the error behavior does seem to work for all of them.

We may not want to use the Rails logger for capturing exceptions, as this information is already  logged elsewhere (you can see the custom exceptions showing up in the "catch all" widget on our migration dashboard such as [this one](https://vagov.ddog-gov.com/dashboard/aq9-bew-hjh/benefits---disability---526---veteran-uploads-lighthouse-migration-staging?fromUser=false&refresh_mode=sliding&from_ts=1731605495012&to_ts=1731609095012&live=true))

The purpose of logging the exception here would just be to have a unified system of logging events in the upload providers, as this logging matches how we log attempts and successes, in addition to the metrics, which are more helpful for aggregating data

Reply:
One option to just get this working is to just increment the metric and not worry about a redundant call to the rails logger